src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,60 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final String COMMA = ",";
+    private int supply;
+    private int buy;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File file = new File(fromFileName);
+        if (file.exists()) {
+            try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+                String readeValue = reader.readLine();
+                while (readeValue != null) {
+                    String[] bufferString = readeValue.split(COMMA);
+                    if (bufferString[0].equals(SUPPLY)) {
+                        supply += Integer.parseInt(bufferString[1]);
+                    }
+                    if (bufferString[0].equals(BUY)) {
+                        buy += Integer.parseInt(bufferString[1]);
+                    }
+                    readeValue = reader.readLine();
+                }
+            } catch (IOException e) {
+                throw new RuntimeException("Can't read file", e);
+            }
+        }
+        String report = createReport();
+        writeToFile(report, toFileName);
+        supply = 0;
+        buy = 0;
+    }
+
+    private void writeToFile(String report, String toFileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            writer.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write to file", e);
```

Comment:
![image](https://github.com/mate-academy/jv-work-with-file/assets/20326858/63b79d0b-296f-49bc-bd76-b0833a8dc0ad)

Reply:
File name added to exception

Reply:
I still do not see File name
![image](https://github.com/mate-academy/jv-work-with-file/assets/20326858/0a30caba-51a4-45ce-b5dd-eacdd26c249d)